pallets/identity/src/keys.rs

```diff
@@ -423,6 +431,152 @@ impl<T: Config> Module<T> {
         Ok(())
     }
 
+    /// Create a child identity.
+    pub(crate) fn base_create_child_identity(
+        origin: T::RuntimeOrigin,
+        secondary_key: T::AccountId,
+    ) -> DispatchResult {
+        let (_, parent_did) = Self::ensure_primary_key(origin)?;
+
+        // Make sure `parent_did` has no parent.
+        Self::ensure_no_parent(parent_did)?;
+
+        // Ensure that the key is a secondary key.
+        Self::ensure_secondary_key(parent_did, &secondary_key)?;
+
+        // Ensure that the key can be unlinked.
+        Self::ensure_key_unlinkable_from_did(&secondary_key)?;
+
+        // Generate a new DID for the child.
+        let child_did = Self::make_did()?;
+
+        // Remove links and get all authorization IDs per signer.
+        // Unlink the secondary account key.
+        Self::remove_key_record(&secondary_key, Some(parent_did));
```

Comment:
@Neopallium would it be worth wrapping up some of this functionality (removing key from an identity) into a cover function to ensure more consistent behaviour across the different use-cases?